src/Microsoft.Health.ControlPlane.Core/Features/Rbac/RbacService.cs

```diff
@@ -23,16 +25,35 @@ public RbacService(IControlPlaneDataStore controlPlaneDataStore)
 
         public async Task<IdentityProvider> GetIdentityProviderAsync(string name, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
-
             return await _controlPlaneDataStore.GetIdentityProviderAsync(name, cancellationToken);
         }
 
         public async Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider identityProvider, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNull(identityProvider, nameof(identityProvider));
-
             return await _controlPlaneDataStore.UpsertIdentityProviderAsync(identityProvider, cancellationToken);
         }
+
+        public async Task<Role> GetRoleAsync(string name, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
+            return await _controlPlaneDataStore.GetRoleAsync(name, cancellationToken);
+        }
+
+        public async Task<IEnumerable<Role>> GetRoleForAllAsync(CancellationToken cancellationToken)
+        {
+            return await _controlPlaneDataStore.GetRoleAllAsync(cancellationToken);
+        }
+
+        public async Task<Role> UpsertRoleAsync(Role role, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(role, nameof(role));
+            return await _controlPlaneDataStore.UpsertRoleAsync(role, cancellationToken);
+        }
+
+        public async Task<string> DeleteRoleAsync(string name, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
+            return await _controlPlaneDataStore.DeleteRoleAsync(name, cancellationToken);
```

Comment:
Similarly here we should check that this isn't the last role before removing it. If it is the last role, we should error out and not allow it. This may be easiest handled by a stored proc because we'll also need to clean up role memberships once they are implemented. Examples of stored procs are in the Fhir control plane.